packages/bierzo-wallet/src/routes/login/index.e2e.spec.ts

```diff
@@ -42,69 +42,70 @@ withChainsDescribe(
-      await page.bringToFront();
-      await travelToWelcomeE2e(page);
-      await sleep(1000);
-      await rejectGetIdentitiesRequest(extensionPage);
-      await page.bringToFront();
-      await checkLoginMessage(page);
-    }, 45000);
-
-    it('shows login to IOV extension if not persona detected', async (): Promise<void> => {
-      browser = await launchBrowser();
-      page = await createPage(browser);
-      await getBackgroundPage(browser);
-      await page.bringToFront();
-      await sleep(1000);
-
-      await page.click('button');
-      await sleep(500);
-
-      await checkLoginMessage(page);
-    }, 45000);
+    withChainsDescribe(
+      'E2E > Login route',
+      (): void => {
+        beforeEach(async (): Promise<void> => {
+          browser = await launchBrowser();
+          page = await createPage(browser);
+          extensionPage = await createExtensionPage(browser);
+        }, 45000);
+
+        afterEach(
+          async (): Promise<void> => {
+            await closeBrowser(browser);
+          },
+        );
+
+        async function checkLoginMessage(page: Page): Promise<void> {
+          const element = await page.$('h6');
+          if (element === null) {
+            throw new Error();
+          }
+          const text = await (await element.getProperty('textContent')).jsonValue();
+          expect(text).toBe(LOGIN_EXTENSION_MSG);
+
+          await closeToast(page);
+        }
+
+        it('should redirect when enqueued login request is accepted', async (): Promise<void> => {
+          await getBackgroundPage(browser);
+          await submitExtensionSignupForm(extensionPage, 'username', '12345678');
+          await page.bringToFront();
+          await travelToWelcomeE2e(page);
+          await sleep(1000);
+          await acceptGetIdentitiesRequest(extensionPage);
+          await page.bringToFront();
+          await whenOnNavigatedToE2eRoute(page, BALANCE_ROUTE);
+        }, 45000);
+
+        it('should stay in login view if enqueued login request is rejected', async (): Promise<void> => {
+          await getBackgroundPage(browser);
+          await submitExtensionSignupForm(extensionPage, 'username', '12345678');
+          await page.bringToFront();
+          await travelToWelcomeE2e(page);
+          await sleep(1000);
+          await rejectGetIdentitiesRequest(extensionPage);
+          await page.bringToFront();
+          await checkLoginMessage(page);
+        }, 45000);
+
+        it('shows login to IOV extension if not persona detected', async (): Promise<void> => {
+          await getBackgroundPage(browser);
+          await page.bringToFront();
+          await sleep(1000);
+
+          await page.click('button');
+          await sleep(500);
+
+          await checkLoginMessage(page);
+        }, 45000);
+      },
+    );
 
     it('shows install IOV extension message', async (): Promise<void> => {
       browser = await launchBrowser(0, false);
```

Comment:
Because we are launching browser here we should close previous instance of browser, so I was required to refactor tests suite a little and use different blocks for launching browser.